exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -317,3 +324,259 @@ func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, clu
+		return err
+	}
+
+	setOfInfraRefNames := map[string]struct{}{}
+	for _, machine := range machineList.Items {
+		infraRef := machine.Spec.InfrastructureRef
+		setOfInfraRefNames[infraRef.Name] = struct{}{}
+	}
+
+	for i := range infraMachines {
+		infraMachine := &infraMachines[i]
+		if _, infraRefExists := setOfInfraRefNames[infraMachine.GetName()]; infraRefExists {
+			continue
+		}
+
+		machine := getNewMachine(mp, infraMachine, false)
+		if err := r.Client.Create(ctx, machine); err != nil {
+			return errors.Wrapf(err, "failed to create new Machine for infraMachine %q in namespace %q", infraMachine.GetName(), infraMachine.GetNamespace())
+		}
+
+		// Set the owner reference on the infraMachine to the Machine since the infraMachine is created and owned by the infraMachinePool.
+		infraMachine.SetOwnerReferences([]metav1.OwnerReference{
+			*metav1.NewControllerRef(machine, machine.GroupVersionKind()),
+		})
+
+		patchHelper, err := patch.NewHelper(infraMachine, r.Client)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create patch helper for infraMachine %q in namespace %q", infraMachine.GetName(), infraMachine.GetNamespace())
+		}
+
+		if err := patchHelper.Patch(ctx, infraMachine); err != nil {
+			return errors.Wrapf(err, "failed to patch infraMachine %q in namespace %q", infraMachine.GetName(), infraMachine.GetNamespace())
+		}
+	}
+	return nil
+}
+
+// reconcileFallbackMachines creates Machines for MachinePools that do not support MachinePool Machines.
+// These Machines will contain only basic information and exist to make a more consistent user experience across all MachinePools and MachineDeployments and do not have an infraRef.
+func (r *MachinePoolReconciler) reconcileFallbackMachines(ctx context.Context, mp *expv1.MachinePool) error {
+	machineList := &clusterv1.MachineList{}
+	labels := map[string]string{
+		expv1.MachinePoolNameLabel:      mp.Name,
+		clusterv1.ClusterNameLabel:      mp.Spec.ClusterName,
+		clusterv1.MachinePoolOwnedLabel: "true",
+	}
+
+	if err := r.Client.List(ctx, machineList, client.InNamespace(mp.Namespace), client.MatchingLabels(labels)); err != nil {
+		return err
+	}
+
+	numMachines := len(machineList.Items)
+
+	if mp.Spec.Replicas == nil {
+		return errors.New("MachinePool.Spec.Replicas is nil, this is unexpected")
+	}
+
+	for numMachines > int(*mp.Spec.Replicas) {
+		machine := &machineList.Items[numMachines-1]
+		if err := r.Client.Delete(ctx, machine); err != nil {
+			return errors.Wrapf(err, "failed to delete Machine %q in namespace %q", machineList.Items[numMachines-1].Name, machineList.Items[numMachines-1].Namespace)
+		}
+		numMachines--
+	}
+
+	for numMachines < int(*mp.Spec.Replicas) {
+		machine := getNewMachine(mp, nil, true)
+		if err := r.Client.Create(ctx, machine); err != nil {
+			return errors.Wrap(err, "failed to create new Machine")
+		}
+		numMachines++
+	}
+
+	return nil
+}
+
+func setMachineLabel(machine *clusterv1.Machine, key string) {
```

Comment:
this seems like a pretty generic helper, I wonder if we already have something like it in utils to set a label on an object that we could reuse instead

Reply:
I don't think so, I originally just manually set the values to true, i.e. writing `machine.Labels[clusterv1.MachinePoolOwnedLabel] = "true"` but the linter was complaining about `"true"` not being a constant.

Reply:
This is the existing pattern in other places in the code: https://github.com/kubernetes-sigs/cluster-api/blob/main/internal/controllers/cluster/cluster_controller_phases.go#L111

https://github.com/kubernetes-sigs/cluster-api/blob/main/util/labels/helpers.go seems like a good place to put this generic helper (and make it truly generic so it can be used for any `o metav1.Object`, not just `Machines`). Similar to https://github.com/kubernetes-sigs/cluster-api/blob/main/util/annotations/helpers.go#L67

Reply:
Makes sense, just made that change. For the sake of scope, I can do a follow up PR to use the helper in other controllers as well.